pkg/chunk/table_manager.go

```diff
@@ -72,6 +78,8 @@ type ProvisionConfig struct {
 // RegisterFlags adds the flags required to config this to the given FlagSet.
 func (cfg *TableManagerConfig) RegisterFlags(f *flag.FlagSet) {
 	f.BoolVar(&cfg.ThroughputUpdatesDisabled, "table-manager.throughput-updates-disabled", false, "If true, disable all changes to DB capacity")
+	f.BoolVar(&cfg.RetentionDeletesEnabled, "table-manager.retention-deletes-enabled", false, "If true, enables retention deletes of DB tables")
```

Comment:
Why do we need both (a boolean flag to enable) and (0 duration to disable) ?

Reply:
The thought behind that was that you could set retention and keep deletes off to verify in the logs that it wants to delete the correct tables, before actually turning on the actual deletes.